Common/Python/PandasConverter.cs

```diff
@@ -66,15 +66,32 @@ public PyObject GetDataFrame(IEnumerable<Slice> data)
 
             foreach (var slice in data)
             {
-                foreach (var baseData in slice.Values)
+                foreach (var key in slice.Keys)
```

Comment:
from a performance standpoint it's much better to iteration directly over the dictionary which will yield `KeyValuePair<TKey, Value>`  -- this prevents us from needing to re-index into the dictionary every time we want the value.